src/averageAges.js

```diff
@@ -1,64 +1,79 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+const MALE = 'm';
+const FEMALE = 'f';
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const selectedMenGroup = people
+    .filter(person => person.sex === MALE
+      && (century === undefined || century === Math.ceil(person.died / 100)));
+  const menAverageAge = selectedMenGroup
+    .reduce((ageSum, person) => ageSum + person.died - person.born, 0);
```

Comment:
Maybe we can create a function for this, we do same calculation twice, for men and women

Reply:
Thank you Dmytro, this is undoubtfully so.